File: div3/1016/C.py
def isPrime(x) :
	x = int(x)
	if x in (1, 0) : return False
	i = 2
	while(i * i <= x) :
		if x % i == 0 :
			return False
		i += 1
	return True

for _ in range(int(input())):
	x, k = map(int, input().split())
	if x == 1:
		x = str(x) * k
		if isPrime(x) : print("YES")
		else : print("NO")
	else:
		if k > 1 : print("NO")
		else :
			if isPrime(x) : print("YES")
			else: print("NO")

File: div3/1016/C.cpp
#include<bits/stdc++.h>
#define int long long
using namespace std;

bool isPrime(int x) {
	if (x == 1 || x == 0) return 0;
	for (int i = 2; i * i <= x; i++) {
		if (x % i == 0) return 0;
	}
	return 1;
}

void solve() {
	string s; int k; 
	cin >> s >> k;
	string t;
	if (stoll(s) == 1) {
		for (int i=0; i<k; i++) {
			t += s;
		}
		s = t;
		int temp = stoll(s);
		if (isPrime(temp)) {
			cout << "YES\n";
		} else {
			cout << "NO\n";
		}
		return;
	}
	if (k > 1) {
		cout << "NO\n";
	} else {
		if (isPrime(stoll(s))) {
			cout << "YES\n";
		} else {
			cout << "NO\n";
		}
	}
}

int32_t main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);cout.tie(nullptr);
	int t; cin >> t;
	while(t--) solve();
}
